components/eam/src/physics/crm/samxx/setperturb.cpp

```diff
@@ -1,47 +1,39 @@
-
 #include "setperturb.h"
-
 void setperturb() {
   // Add random noise near the surface to help turbulence develop
-  // This surboutine has been updated for SPCAM5 (Minghuai.Wang@pnnl.gov, April, 2012).
-  // Now the random generator is seeded based on the global column id, which gets rid
-  // of the dependence of the SPCAM results on pcols.
-  // Walter Hannah - LLNL - Mar 2018
+  // The random generator is seeded based on the global column id, 
+  // which avoids a problematic sensitivity to pcols.
   int  constexpr perturb_num_layers  = 5;    // Number of levels to perturb
   real constexpr perturb_t_magnitude = 1.0;  // perturbation LSE amplitube [K]
-
-  real2d t02 = real2d("t02",perturb_num_layers,ncrms);
-  yakl::memset(t02,0.);
-
+  real factor_xy = 1. / (nx*ny);
+  auto t_host     = t    .createHostCopy();
+  auto t0_host    = t0   .createHostCopy();
```

Comment:
Formatting question; do mean for there to be extra spaces before the "."?

Reply:
Only if it allows `.createHostCopy()` to line up with other variables above and below this code snippet.

Reply:
I'll lose the spacing.